net/devif/devif_send.c

```diff
@@ -102,7 +102,12 @@ void devif_send(FAR struct net_driver_s *dev, FAR const void *buf,
 
   /* Prepare device buffer before poll callback */
 
-  iob_update_pktlen(dev->d_iob, offset);
+  /* if pktlen is 0, no need to update */
+
+  if (offset != 0)
```

Comment:
since `iob->io_pktlen` is 0 anyway from https://github.com/tiiuae/nuttx/blob/07e56328f685c3ea23a3a44e9a973f3f7eb44c32/mm/iob/iob_alloc.c#L304
in the `netdev_iob_prepare` before this. There is no need for iob_update_pktlen is offset is also 0